consensus/types/src/data_column_sidecar.rs

```diff
@@ -121,13 +137,62 @@ impl<T: EthSpec> DataColumnSidecar<T> {
         Ok(column)
     }
 
+    pub fn id(&self) -> DataColumnIdentifier {
+        DataColumnIdentifier {
+            block_root: self.block_root(),
+            index: self.index,
+        }
+    }
+
     pub fn slot(&self) -> Slot {
         self.signed_block_header.message.slot
     }
 
     pub fn block_root(&self) -> Hash256 {
         self.signed_block_header.message.tree_hash_root()
     }
+
+    pub fn min_size() -> usize {
+        // min size is one cell
+        Self {
+            index: 0,
+            column: VariableList::new(vec![Cell::<T>::default()]).unwrap(),
+            kzg_commitments: VariableList::new(vec![KzgCommitment::empty_for_testing()]).unwrap(),
+            kzg_proofs: VariableList::new(vec![KzgProof::empty()]).unwrap(),
+            signed_block_header: SignedBeaconBlockHeader {
+                message: BeaconBlockHeader::empty(),
+                signature: Signature::empty(),
+            },
+            kzg_commitments_inclusion_proof: Default::default(),
+        }
+        .as_ssz_bytes()
+        .len()
+    }
+
+    pub fn max_size() -> usize {
+        Self {
+            index: 0,
+            column: VariableList::new(vec![Cell::<T>::default(); T::MaxBlobsPerBlock::to_usize()])
+                .unwrap(),
+            kzg_commitments: VariableList::new(vec![
+                KzgCommitment::empty_for_testing();
```

Comment:
The sizes the same here? Empty and non-empty?

Reply:
The "empty" `KzgCommitment` and `KzgProof` means a fixed size array filled with `0`s. So here it's filling the `Vec`s with 4096 (`MaxBlobCommitmentsPerBlock`) instances of empty `KzgCommitment`s and `KzgProof`s (0 filled fixed-size array).

Although I think we won't actually have `MaxBlobCommitmentsPerBlock` number of items here - this upper bound used in the type was to make future upgrades easier but in practice this would be bound to `MaxBlobsPerBlock` instead. I'll update this.

Reply:
Updated in https://github.com/sigp/lighthouse/pull/5196/commits/3a967847ece611ee5486b4061a8b13434b383258